tests/upgrade_path/upgrade_helpers.py

```diff
@@ -167,3 +170,52 @@ def get_copp_cfg_formatted_dict(copp_cfg, feature_status):
     formatted_dict.update({"default": copp_cfg["COPP_GROUP"]["default"]})
     logging.debug("Formatted copp_cfg.json dictionary: {}".format(formatted_dict))
     return formatted_dict
+
+
+def upgrade_test_helper(duthost, localhost, ptfhost, from_image, to_image,
+                        tbinfo, upgrade_type, get_advanced_reboot,
+                        advanceboot_loganalyzer, modify_reboot_script=None, allow_fail=False,
+                        sad_preboot_list=None, sad_inboot_list=None, reboot_count=1,
+                        enable_cpa=False, preboot_setup=None, postboot_setup=None):
+
+    reboot_type = get_reboot_command(duthost, upgrade_type)
+    if enable_cpa and "warm-reboot" in reboot_type:
```

Comment:
A knob to enable/disable this support for CPA is missing in this change.

Reply:
Added, and verified that ferret was started in the PTF container.